ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdated.java

```diff
@@ -119,7 +119,8 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     }
 
     // TODO: post-merge cleanup, this should be unnecessary after merge
-    if (!mergeCoordinator.latestValidAncestorDescendsFromTerminal(newHead)) {
+    if (!mergeContext.get().isNearHeadCheckpointSync()
```

Comment:
I tried that, and it was way too many changes for what this check is doing atm and IMO completely unnecessary since this is a validation that we're already planning to disable in a near future. This is only kept because some of the testnets apparently.